Old issue needs to be properly addressed

The import line
`import { Web3Storage, getFilesFromPath } from 'web3.storage';`
leads to the error
_Module not found: Can't resolve 'ipfs-car/blockstore/memory'_
on ReactJS.

I know this issue has been mentioned here #312 and here #343 but it seems it hasn't been solved yet.

Same error.
`Module not found: Can't resolve 'ipfs-car/blockstore/memory'` 

when added `import { Web3Storage } from 'web3.storage/dist/bundle.esm.min.js'` getting heap out of memory error

Ok, guys I finally was able to fix the problem for me, so I paste my solution here hoping to help people in need.
**For the team**
Looks to me that if you remap the package export routes everything can run clean for all of us, if this is true please let me know because i'm flagging the whole package in order not to update it and keep my modifications to the map on the final build.
**For people**
Simple steps:
1_update to webpack 5, I followed [this tut](https://dev.to/riyanegi/setting-up-webpack-5-with-react-and-babel-from-scratch-2021-271l) 
2_Update the package.json of your project AND all routes in the webpack config.
3_create an npm "serve" script running both npm build and webpack serve command as explained in the tut.
4 _call the component with `import { Web3Storage } from 'web3.storage/dist/bundle.esm.min.js'` in your App.js

This should do the magic for dev environment, when you build for production it throws an error:

`Module not found: Error: Package path ./dist/bundle.esm.min.js is not exported from package C:\Users\jeelt\source\repos\webpack5\node_modules\web3.storage (see exports field in C:\Users\jeelt\source\repos\webpack5\node_modules\web3.storage\package.json)`

And here's the problem with the map.
I changed the exports map with the suggested package route directly into the node_module folder web3.storage, name of file is package.json, that's why i won't be able to update anymore without losing this change, needed for the final react build. this is how my exports look like now:

`"exports": {
    ".": {
      "browser": "./src/lib.js",
      "require": "./dist/src/lib.cjs",
      "node": "./src/lib.js"
	  
    },
    "./src/platform.js": {
      "browser": "./src/platform.web.js",
      "require": "./dist/src/platform.cjs",
      "node": "./src/platform.js"
    },
    "./name": {
      "browser": "./src/name.js",
      "require": "./dist/src/name.cjs",
      "node": "./src/name.js"
    },
	"./dist/bundle.esm.min.js": {
		"browser": "./dist/bundle.esm.min.js",
		"require": "./dist/bundle.esm.min.js",
		"node": "./dist/bundle.esm.min.js"
	}
  },`
Hope this helps someone, or _if I'm doing wrong hope someone from the team will tell me to stop immediately because I'm doing something very dangerous that i didn't recognise_.
Cheers

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -72,22 +72,26 @@
   "types": "./dist/src/lib.d.ts",
   "module": "./src/lib.js",
   "main": "./dist/src/lib.cjs",
-  "exports": {
-    ".": {
-      "browser": "./src/lib.js",
-      "require": "./dist/src/lib.cjs",
-      "node": "./src/lib.js"
-    },
-    "./src/platform.js": {
-      "browser": "./src/platform.web.js",
-      "require": "./dist/src/platform.cjs",
-      "node": "./src/platform.js"
-    },
-    "./name": {
-      "browser": "./src/name.js",
-      "require": "./dist/src/name.cjs",
-      "node": "./src/name.js"
-    }
+  "exports": { 
+    ".": { 
+      "browser": "./src/lib.js", 
+      "require": "./dist/src/lib.cjs", 
+      "node": "./src/lib.js" 
+    }, 
+    "./src/platform.js": { 
+      "browser": "./src/platform.web.js", 
+      "require": "./dist/src/platform.cjs", 
+      "node": "./src/platform.js" 
+    }, 
+    "./name": { 
+      "browser": "./src/name.js", 
+      "require": "./dist/src/name.cjs", 
+      "node": "./src/name.js" }, 
+    "./dist/bundle.esm.min.js": { 
+      "browser": "./dist/bundle.esm.min.js", 
+      "require": "./dist/bundle.esm.min.js", 
+      "node": "./dist/bundle.esm.min.js" 
+      }
   },
   "browser": {
     "./src/platform.js": "./src/platform.web.js"

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -72,20 +72,20 @@
   "types": "./dist/src/lib.d.ts",
   "module": "./src/lib.js",
   "main": "./dist/src/lib.cjs",
-  "exports": { 
-    ".": { 
-      "browser": "./src/lib.js", 
-      "require": "./dist/src/lib.cjs", 
-      "node": "./src/lib.js" 
+  "exports": {
+    ".": {
+      "browser": "./src/lib.js",
+      "require": "./dist/src/lib.cjs",
+      "node": "./src/lib.js"
     }, 
-    "./src/platform.js": { 
-      "browser": "./src/platform.web.js", 
-      "require": "./dist/src/platform.cjs", 
-      "node": "./src/platform.js" 
-    }, 
-    "./name": { 
-      "browser": "./src/name.js", 
-      "require": "./dist/src/name.cjs", 
+    "./src/platform.js": {
+      "browser": "./src/platform.web.js",
+      "require": "./dist/src/platform.cjs",
+      "node": "./src/platform.js"
+    },
+    "./name": {
+      "browser": "./src/name.js",
+      "require": "./dist/src/name.cjs",
       "node": "./src/name.js"
     }, 
     "./dist/bundle.esm.min.js": { 

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -86,12 +86,13 @@
     "./name": { 
       "browser": "./src/name.js", 
       "require": "./dist/src/name.cjs", 
-      "node": "./src/name.js" }, 
+      "node": "./src/name.js"
+    }, 
     "./dist/bundle.esm.min.js": { 
       "browser": "./dist/bundle.esm.min.js", 
       "require": "./dist/bundle.esm.min.js", 
       "node": "./dist/bundle.esm.min.js" 
-      }
+    }
   },
   "browser": {
     "./src/platform.js": "./src/platform.web.js"

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -77,7 +77,7 @@
       "browser": "./src/lib.js",
       "require": "./dist/src/lib.cjs",
       "node": "./src/lib.js"
-    }, 
+    },
     "./src/platform.js": {
       "browser": "./src/platform.web.js",
       "require": "./dist/src/platform.cjs",

